backend/tarsy/services/parallel_stage_executor.py

```
@@ -0,0 +1,1005 @@
+        Returns:
+            True if the last stage is parallel, False otherwise
+        """
+        if not chain_definition.stages:
+            return False
+        
+        last_stage = chain_definition.stages[-1]
+        return last_stage.agents is not None or last_stage.replicas > 1
+    
+    async def synthesize_parallel_results(
+        self,
+        parallel_result: ParallelStageResult,
```

Ruff (0.14.8)

Unused method argument: `parallel_result`

(ARG002)